app/src/main/java/in/testpress/testpress/ui/AttemptQuestionsFragment.java

```diff
@@ -258,18 +307,30 @@ public void createCheckBoxView(final List<AttemptAnswer> attemptAnswers, Attempt
             option.setId(i);
             LayoutParams layoutParams = new LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT);
             option.setLayoutParams(layoutParams);
-            Spanned html = Html.fromHtml(answer, new AnswerImageGetter(), null);
+            Spanned html = Html.fromHtml(answer, new AnswerImageGetter(i), null);
+            final int noOfImages = answerImagesUrl.get(i + "").size();
             option.setText(trim(html, 0, html.length()));
-            for(int j = 0; j < answerImagesUrl.size() ; j++) {
-                imageLoader.loadImage(answerImagesUrl.get(j), new SimpleImageLoadingListener() {
-                    @Override
-                    public void onLoadingComplete(String imageUri, View view, Bitmap loadedImage) {
-                        Drawable drawable = new BitmapDrawable(loadedImage);
-                        answerImages.put(imageUri, drawable);
-                        Spanned html = Html.fromHtml(answer, new AnswerImageSetter(), null);
-                        option.setText(trim(html, 0, html.length()));
-                    }
-                });
+            for(int j = 0; j < noOfImages ; j++) {
+                if (answerImages.containsKey(answerImagesUrl.get(i + "").get(j))) {
+                        html = Html.fromHtml(answer, new AnswerImageSetter(), null);
+                        option.setText(clickableText(html));
+                        option.setMovementMethod(LinkMovementMethod.getInstance());
+                } else {
+                    imageLoader.loadImage(answerImagesUrl.get(i + "").get(j), new SimpleImageLoadingListener() {
+                        @Override
+                        public void onLoadingComplete(String imageUri, View view, Bitmap loadedImage) {
+                            Drawable drawable = new BitmapDrawable(loadedImage);
+                            answerImages.put(imageUri, drawable);
+                            noOfImagesLoaded++;
```

Comment:
The noOfImages variable is per answer. While the noOfImagesLoaded is a class variable and hence is common for all answers. Shouldn't it be per answer instead of class variable? Is this code tested with 
answer 1 with 5 image, answer 2 with 1 images, answer 3 with 5 images etc? The logic is prone to fail since noOfImagesLoaded is a class variable. But noOfImages is a local variable. Also please name the variables properly. noOfImages means what images? noOfImagesLoaded means what images? The name of the variable should explain what it is used for.